src/transports/dash/manifest_parser.ts

```diff
@@ -154,34 +158,105 @@ export default function generateManifestParser(
                       responseType: "arraybuffer" as const,
                       cancelSignal });
           return req;
+        }).then((res) => {
+          if (value.format === "string") {
+            if (typeof res.responseData !== "string") {
+              throw new Error("External DASH resources should have been a string");
+            }
```

Comment:
Now that I think of this, this check / error throwing was only there before so TypeScript would understand that it cannot be an ArrayBuffer / a string.
But now I'm wondering if this check is still necessary (we added a runtime check later anyway)

Reply:
Unfortunately, it seems that typescript is still yelling if I remove the check.